crates/ruff_linter/src/rules/flake8_pytest_style/rules/assertion.rs

```diff
@@ -364,9 +367,84 @@ impl Violation for PytestUnittestRaisesAssertion {
 }
 
 /// PT027
-pub(crate) fn unittest_raises_assertion(
+pub(crate) fn unittest_raises_assertion_call(checker: &mut Checker, call: &ast::ExprCall) {
+    // Bindings in `with` statements are handled by `unittest_raises_assertion_with`.
+    if let Stmt::With(ast::StmtWith { items, .. }) = checker.semantic().current_statement() {
+        if items
+            .iter()
+            .any(|item| item.context_expr.range() == call.range && item.optional_vars.is_some())
```

Comment:
Do I understand it correctly that what we want to test if `context_expr` is the `call`? If so, I suggest using `AnyNodeRef::ptr_eq`, it makes it more explicit rather than comparing ranges.